src/Diffusion.jl

```diff
@@ -108,4 +113,73 @@ function nmolabsorb(σ, steps, pos, r, molecules) #simulates multiple molecules
     return molhistories
 end
 
+### Analysis ###
+ 
+function molremaining(trial)
+    lengths = molpathlengths(trial)
+    return percentremaining(lengths)
+end
+    
+function molpathlengths(trial) #input the molhistories given my nmolreflect or nmolabsorb
+    lengthsofmolpaths = Int[]         #outputs a vector showing the lengths of each molecules' path
+    for i in 1:length(trial)
+        push!(lengthsofmolpaths, length(trial[i]))
+    end
+    return lengthsofmolpaths
+end
+    
+    
+function numberremaining(lengthsofmolpaths) #input a vector containing the length of each molecules path
+    numberremain = Int[]                       #out puts a vector containing the number of molecules remaining each time step
+        for x in 1:maximum(lengthsofmolpaths)
+            number = 0.0
```

Comment:
```suggestion
            number = 0
```
so it stays an `Int`.